test/e2e/framework/test_context.go

```diff
@@ -18,82 +18,56 @@ package framework
-//
-// Test suite authors can use framework/viper to make all command line
-// parameters also configurable via a configuration file.
 type TestContextType struct {
+	KubeHost    string
 	KubeConfig  string
 	KubeContext string
-	Host        string
 
-	OutputDir string
+	ReportDir string
+
+	ChartPath   string
+	ChartValues string
 
-	// If set to true test will dump data about the namespace in which test was running.
+	OperatorImageTag string
+	SidecarImage     string
+
+	TimeoutSeconds    int
 	DumpLogsOnFailure bool
-	// Disables dumping cluster log from master and nodes after all tests.
-	DisableLogDump bool
-	TimeoutSeconds int
 }
 
-// TestContext should be used by all tests to access common context data.
 var TestContext TestContextType
 
-// RegisterCommonFlags registers flags common to all e2e test suites.
-// The flag set can be flag.CommandLine (if desired) or a custom
-// flag set that then gets passed to viperconfig.ViperizeFlags.
-//
-// The other Register*Flags methods below can be used to add more
-// test-specific flags. However, those settings then get added
-// regardless whether the test is actually in the test suite.
-//
-// For tests that have been converted to registering their
-// options themselves, copy flags from test/e2e/framework/config
-// as shown in HandleFlags.
-func RegisterCommonFlags(flags *flag.FlagSet) {
-	flags.StringVar(&TestContext.KubeConfig, clientcmd.RecommendedConfigPathFlag, os.Getenv(clientcmd.RecommendedConfigPathEnvVar), "Path to kubeconfig containing embedded authinfo.")
-	flags.StringVar(&TestContext.KubeContext, clientcmd.FlagContext, "", "kubeconfig context to use/override. If unset, will use value from 'current-context'")
-	flags.StringVar(&TestContext.Host, "host", "", fmt.Sprintf("The host, or apiserver, to connect to. Will default to %s if this argument and --kubeconfig are not set.", defaultHost))
-	flags.StringVar(&TestContext.OutputDir, "e2e-output-dir", "/tmp", "Output directory for interesting/useful test data, like performance data, benchmarks, and other metrics.")
-	flags.BoolVar(&TestContext.DumpLogsOnFailure, "dump-logs-on-failure", true, "If set to true test will dump data about the namespace in which test was running.")
-	flags.BoolVar(&TestContext.DisableLogDump, "disable-log-dump", false, "If set to true, logs from master and nodes won't be gathered after test run.")
-	flag.IntVar(&TestContext.TimeoutSeconds, "pod-wait-timeout", 100, "Timeout to wait for a pod to be ready.")
+// Register flags common to all e2e test suites.
+func RegisterCommonFlags() {
+	// Turn on verbose by default to get spec names
+	config.DefaultReporterConfig.Verbose = true
+
+	// Turn on EmitSpecProgress to get spec progress (especially on interrupt)
+	config.GinkgoConfig.EmitSpecProgress = true
+
+	// Randomize specs as well as suites
+	config.GinkgoConfig.RandomizeAllSpecs = true
+
+	flag.StringVar(&TestContext.KubeHost, "kubernetes-host", "", "The kubernetes host, or apiserver, to connect to")
```

Comment:
use `fmt.Sprintf("%s:%s", os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")) ` may be better

Reply:
Kubeconfig contains Kubehost configuration, so when the remote connection k8s, the Host in kubeconfig can be used directly, no need to fill in the kubehost parameters.